mitiq/lre/tests/test_multivariate_richardson.py

```diff
@@ -0,0 +1,239 @@
+    calculated_variables = _create_variable_combinations(
+        test_num_layers, test_degree
+    )
+    assert len(calculated_variables) == comb(
+        test_num_layers + test_degree, test_degree
+    )
+
+    # check last element is an empty tuple to account for degree = 0
+    assert not calculated_variables[-1]
+
+
+@pytest.mark.parametrize(
+    "test_num_layers, test_degree, expected_basis",
+    [
+        (2, 2, ["1", "λ_2", "λ_1", "λ_2**2", "λ_1*λ_2", "λ_1**2"]),
+        (
+            3,
+            2,
+            [
+                "1",
+                "λ_3",
+                "λ_2",
+                "λ_1",
+                "λ_3**2",
+                "λ_2*λ_3",
+                "λ_2**2",
+                "λ_1*λ_3",
+                "λ_1*λ_2",
+                "λ_1**2",
+            ],
+        ),
+    ],
+)
+def test_full_monomial_basis(test_num_layers, test_degree, expected_basis):
+    calculated_basis = full_monomial_basis(test_num_layers, test_degree)
+    assert len(calculated_basis) == comb(
+        test_num_layers + test_degree, test_degree
+    )
+
+    assert calculated_basis == expected_basis
+
+
+qreg1 = LineQubit.range(3)
+test_circuit1 = Circuit(
+    [ops.H.on_each(*qreg1)],
+    [ops.CNOT.on(qreg1[0], qreg1[1])],
+)
+test_circuit2 = Circuit(
+    [ops.H.on_each(*qreg1)],
+    [ops.CNOT.on(qreg1[0], qreg1[1])],
+    [ops.X.on(qreg1[2])],
+    [ops.TOFFOLI.on(*qreg1)],
+)
+
+
+@pytest.mark.parametrize(
+    "test_circ, test_degree, expected_matrix",
+    [
+        (
+            test_circuit1,
+            2,
+            np.array(
+                [
+                    [1.0, 1.0, 1.0, 1.0, 1.0, 1.0],
+                    [1.0, 1.0, 3.0, 1.0, 3.0, 9.0],
+                    [1.0, 3.0, 1.0, 9.0, 3.0, 1.0],
+                    [1.0, 1.0, 5.0, 1.0, 5.0, 25.0],
+                    [1.0, 3.0, 3.0, 9.0, 9.0, 9.0],
+                    [1.0, 5.0, 1.0, 25.0, 5.0, 1.0],
+                ]
+            ),
+        ),
+        (
+            test_circuit2,
+            2,
+            np.array(
+                [
+                    [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0],
+                    [1.0, 1.0, 1.0, 3.0, 1.0, 1.0, 1.0, 3.0, 3.0, 9.0],
+                    [1.0, 1.0, 3.0, 1.0, 1.0, 3.0, 9.0, 1.0, 3.0, 1.0],
+                    [1.0, 3.0, 1.0, 1.0, 9.0, 3.0, 1.0, 3.0, 1.0, 1.0],
+                    [1.0, 1.0, 1.0, 5.0, 1.0, 1.0, 1.0, 5.0, 5.0, 25.0],
+                    [1.0, 1.0, 3.0, 3.0, 1.0, 3.0, 9.0, 3.0, 9.0, 9.0],
+                    [1.0, 3.0, 1.0, 3.0, 9.0, 3.0, 1.0, 9.0, 3.0, 9.0],
+                    [1.0, 1.0, 5.0, 1.0, 1.0, 5.0, 25.0, 1.0, 5.0, 1.0],
+                    [1.0, 3.0, 3.0, 1.0, 9.0, 9.0, 9.0, 3.0, 3.0, 1.0],
+                    [1.0, 5.0, 1.0, 1.0, 25.0, 5.0, 1.0, 5.0, 1.0, 1.0],
+                ]
+            ),
+        ),
+    ],
+)
+def test_sample_matrix(test_circ, test_degree, expected_matrix):
```

Comment:
We may want to include tests for the stability of the numerical stability of the sample matrix calculation as well. Something like:

```py
def test_sample_matrix_numerical_stability():
    large_circuit = Circuit([ops.H.on(LineQubit(i)) for i in range(100)])
    matrix = sample_matrix(large_circuit, 5, 1000)
    assert np.isfinite(matrix).all()
    assert not np.isnan(matrix).any()
```